src/kvstore/NebulaStore.cpp

```diff
@@ -118,43 +118,55 @@ void NebulaStore::loadPartFromDataPath() {
         }
 
         // partIds is the partition in this host waiting to open
-        std::vector<PartitionID> partIds;
-        for (auto& partId : enginePtr->allParts()) {
-          if (!options_.partMan_->partExist(storeSvcAddr_, spaceId, partId).ok()) {
+        std::map<PartitionID, HostAddr> partsLearner;
+        for (auto& [partId, learner] : enginePtr->allParts()) {
+          if (!options_.partMan_->partExist(storeSvcAddr_, spaceId, partId).ok() &&
```

Comment:
you should check if the learner's host is equal to this process's host, only a host that is a learner should not be removed